app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -1049,8 +1065,74 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
         private const val TAG = "MainActivity" // logging tag
         private const val DRAWER_ITEM_ADD_ACCOUNT: Long = -13
         private const val DRAWER_ITEM_ANNOUNCEMENTS: Long = 14
-        const val REDIRECT_URL = "redirectUrl"
-        const val OPEN_DRAFTS = "draft"
+        private const val REDIRECT_URL = "redirectUrl"
+        private const val OPEN_DRAFTS = "draft"
+        private const val ACCOUNT_ID = "accountId"
+        private const val COMPOSE_OPTIONS = "composeOptions"
+        private const val NOTIFICATION_TYPE = "notificationType"
+        private const val NOTIFICATION_TAG = "notificationTag"
+        private const val NOTIFICATION_ID = "notificationId"
+
+        /**
+         * switches the active account to the provided accountId and then stays on MainActivity
+         */
+        @JvmStatic
+        fun accountSwitchIntent(context: Context, accountId: Long): Intent {
+            return Intent(context, MainActivity::class.java).apply {
+                putExtra(ACCOUNT_ID, accountId)
+            }
+        }
+
+        /**
+         * switches the active account to the accountId and takes the user to correct place according to the notification they clicked
+         */
+        @JvmStatic
+        fun openNotificationIntent(context: Context, type: Notification.Type, accountId: Long): Intent {
```

Comment:
Please can all of these `...Intent` methods have a consistent param order; `context` first, then `accountId` (possibly called `tuskyAccountId`, per the earlier comment), and then the params that are specific to that intent.